hibernate-core/src/main/java/org/hibernate/engine/spi/ActionQueue.java

```diff
@@ -1341,12 +1337,11 @@ else if ( type.isComponentType() && value != null ) {
 		}
 
 		private void addToBatch(BatchIdentifier batchIdentifier, AbstractEntityInsertAction action) {
-			List<AbstractEntityInsertAction> actions = actionBatches.get( batchIdentifier );
+			List<AbstractEntityInsertAction> actions = actionBatches.computeIfAbsent(
+					batchIdentifier,
+					k -> new LinkedList<>()
+			);
 
-			if ( actions == null ) {
-				actions = new LinkedList<>();
-				actionBatches.put( batchIdentifier, actions );
-			}
```

Comment:
@Sanne , do you think the above code change is necessary? I can revert it back without problem.